src/instructions.py

```diff
@@ -185,19 +176,58 @@ class SettleFundsParams(NamedTuple):
 
 
 def decode_initialize_market(instruction: TransactionInstruction) -> InitializeMarketParams:
-    raise NotImplementedError("decode_initialize_market not implemented.")
+    """Decode an instialize market instruction and retrieve the instruction params."""
+    verify_instruction_keys(instruction, 9)
+    data = INSTRUCTIONS_LAYOUT.parse(instruction.data)
+    return InitializeMarketParams(
+        market=instruction.keys[0].pubkey,
+        request_queue=instruction.keys[1].pubkey,
+        event_queue=instruction.keys[2].pubkey,
+        bids=instruction.keys[3].pubkey,
+        asks=instruction.keys[4].pubkey,
+        base_vault=instruction.keys[5].pubkey,
+        quote_vault=instruction.keys[6].pubkey,
+        base_mint=instruction.keys[7].pubkey,
+        quote_mint=instruction.keys[8].pubkey,
+        base_lot_size=data.args.base_lot_size,
+        quote_lot_size=data.args.quote_lot_size,
+        fee_rate_bps=data.args.fee_rate_bps,
+        vault_signer_nonce=data.args.vault_signer_nonce,
+        quote_dust_threshold=data.args.quote_dust_threshold,
+        program_id=instruction.program_id,
+    )
```

Comment:
I am thinking if we actually need the decoding part. Since my understanding is that the instruction passing is a one-way street where the client only sends the instruction but doesn't receive it.

Reply:
I think this will still be useful to maintain logs. i.e. examining confirmed trades.